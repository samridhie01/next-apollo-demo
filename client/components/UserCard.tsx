import { Card, CardContent } from "@material-ui/core";
import React from "react";
import styled from "styled-components";

interface User {
    name: string,
    addr: string
}

const StyledCard = styled(Card)`
    width: 300px;
    padding: 5px;
    &:hover{
        background-color: grey;
        cursor: pointer

    }
`

const UserCard: React.FC<User> = ({ name, addr }) => {

    return (
        <StyledCard raised={true} className="user-cards">
            <CardContent>
                {name}
            </CardContent>
            <CardContent>
                {addr}
            </CardContent>
        </StyledCard>

    )
}

export default UserCard;