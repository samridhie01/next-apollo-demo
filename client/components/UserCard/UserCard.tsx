import React from "react";
import { Card, CardContent, Grid, makeStyles } from "@material-ui/core";


const useStyles = makeStyles(theme => ({
    usercard: {
        width: '300px',
        height: '150px',
        padding: '5px',
        margin: '1px',
       "&:hover":{
            backgroundColor: 'grey',
            cursor: 'pointer'
        }
    },
    root:{
       justifyContent: 'center'
    }
  }));

const UserCard: React.FC<User> = ({ name, addr }) => {

    const classes = useStyles();
    return (
        <Grid item className={classes.root}> 
        <Card raised={true} className={classes.usercard} data-testid="user-card">
            <CardContent>
                {name}
            </CardContent>
            <CardContent>
                {addr}
            </CardContent>
        </Card>
        </Grid>
    )
}

export default UserCard;