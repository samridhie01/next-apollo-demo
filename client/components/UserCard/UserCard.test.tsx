import { render } from "@testing-library/react"
import React from "react"
import UserCard from "./UserCard"
import renderer from 'react-test-renderer';

describe('UserCard', ()=>{
    it('renderCorrectly', ()=>{
        const component = renderer.create(<UserCard name="abc" addr="Delhi"/>)
        let tree = component.toJSON();
         expect(tree).toMatchSnapshot();
    })
})