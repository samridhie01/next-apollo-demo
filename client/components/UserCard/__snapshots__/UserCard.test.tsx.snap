// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard renderCorrectly 1`] = `
<div
  className="MuiPaper-root MuiCard-root sc-bdnxRM fjnpmH user-cards MuiPaper-elevation8 MuiPaper-rounded"
  data-testid="user-card"
>
  <div
    className="MuiCardContent-root"
  >
    abc
  </div>
  <div
    className="MuiCardContent-root"
  >
    Delhi
  </div>
</div>
`;
